sap-xmake-build.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash -l
+
+# find this script and establish base directory
+curr_dir=`dirname $0`
+MY_DIR=`cd $curr_dir; pwd`
+echo "[INFO] Executing in ${MY_DIR}"
+
+PACKAGE_NAME=${PACKAGE_NAME:-"scala-pickling"}
+PACKAGE_BRANCH=${PACKAGE_BRANCH:-"0.11.x"}
+BUILD_BRANCH=${BUILD_BRANCH:-"v0.11.x-M1_2.11"}
+ALTISCALE_RELEASE="${ALTISCALE_RELEASE:-5.0.0}"
+
+rm -rf install-dir-*
+rm -rf rpm-dir-*
+
+echo "ok - $(whoami) user is going to build the project"
+echo "ok - fpm is located at $(which fpm)"
+echo "ok - PATH=$PATH"
+echo "ok - BUILD_BRANCH=$BUILD_BRANCH"
+echo "ok - PACKAGE_BRANCH=$PACKAGE_BRANCH"
+
+ret=0
+export DATE_STRING=${DATE_STRING:-$(date -u +%Y%m%d%H%M)}
+RPM_INSTALL_DIR=${MY_DIR}/picklingrpmbuild
+echo "RPM_INSTALL_DIR : $RPM_INSTALL_DIR"
+pushd ${MY_DIR}
+
+export RPM_NAME=`echo scala-pickling-${BUILD_BRANCH}`
+export RPM_DESCRIPTION="scala-pickling library ${BUILD_BRANCH}"
+echo "RPM_NAME : $RPM_NAME"
+echo "RPM_DESCRIPTION : $RPM_DESCRIPTION"
+echo "Current directory : ${pwd}"
+
+##################
+# Packaging  RPM #
+##################
+echo "Packaging RPM"
+export RPM_BUILD_DIR="${RPM_INSTALL_DIR}/usr/sap/spark/controller/"
+# Generate RPM based on where spark artifacts are placed from previous steps
+rm -rf "${RPM_BUILD_DIR}"
+mkdir --mode=0755 -p "${RPM_BUILD_DIR}"
+
+if [[ "$BUILD_BRANCH" == *_2.10 ]] ; then
+  mkdir --mode=0755 -p "${RPM_BUILD_DIR}/lib"
+  cp -rp /imports/scala-pickling_2.10-*.jar $RPM_BUILD_DIR/lib/
+elif [[ "$BUILD_BRANCH" == *_2.11 ]] ; then
+  mkdir --mode=0755 -p "${RPM_BUILD_DIR}/lib_2.11"
+  cp -rp /imports/scala-pickling_2.11-*.jar $RPM_BUILD_DIR/lib_2.11/
+else
+  echo "fatal - unsupported version for $BUILD_BRANCH, can't produce RPM, quitting!"
+  exit -1
+fi
+
+mkdir -p "${RPM_BUILD_DIR}/licenses"
+cp LICENSE "${RPM_BUILD_DIR}/licenses/LICENSE-${RPM_NAME}"
+echo "cp successful"
+
+echo "running mkdir for RPM_INSTALL_DIR"
+mkdir -p ${RPM_INSTALL_DIR}
```

Comment:
the RPM_INSTALL_DIR should be ${RPM_INSTALL_DIR} to be useful ?

Reply:
yap.... that is incorrect in the echo, it should be something like this

```
echo "running mkdir for $RPM_INSTALL_DIR"
```

or

```
echo "running mkdir for ${RPM_INSTALL_DIR}"
```

Reply:
Corrected

Reply:
It is a bit inconsistent in this file that sometimes ${VARIABLE} is used and sometimes $VARIABLE. But this is a style issue and as long as both work I'm okay with this.